plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/actions/RequestCheckEdgeTargetAction.java

```diff
@@ -0,0 +1,69 @@
+/********************************************************************************
+ * Copyright (c) 2023 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+package org.eclipse.glsp.server.actions;
+
+import org.eclipse.glsp.server.types.EdgeTypeHint;
+
+/**
+ * A {@link RequestAction} used to check the validity of an edge being created. This is used
+ * to update creation feedback on the client side, for edges configured with a dynamic {@link EdgeTypeHint}.
+ *
+ * @see EdgeTypeHint#isDynamic()
+ * @see CheckEdgeTargetResultAction
+ */
+public class RequestCheckEdgeTargetAction extends RequestAction<CheckEdgeTargetResultAction> {
```

Comment:
Do we really need the request prefix here?  IMO `CheckEdgeTargetAction` should be fine